### PR TITLE
[exec, lib, INT] Add beam28 to test matrix

### PR DESCRIPTION
**Pull request by shireen-bean**
## Description

Add beam28 to the test matrix to test useage with beam's 2.28.0 release.
Update integration tests to use latest klio supported beam version 2.28.0

## Testing
Ran tox for exec and lib libraries - no code logic changes are implemented.
Ran integration tests with updated dependedencies

## Checklist for PR author(s)

- [x] Format the pull request title like `[cli] Fixes bugs in 'klio job fake-cmd'`.
- [x] Changes are covered by unit tests (no major decrease in code coverage %) and/or integration tests.
- [x] Document any relevant additions/changes in the appropriate spot in `docs/src`.
- [x] For any change that affects users, update the package's changelog in ``docs/src/reference/<package>/changelog.rst``

### CODE REVIEW

File: exec/tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = {py36,py37,py38}-beam{25,26,27},docs,manifest,check-formatting,lint
+envlist = {py36,py37,py38}-beam{25,26,27,28},docs,manifest,check-formatting,lint
```
Comment (DanSimon) [inline exec/tox.ini:2]: I believe our policy is only supporting 3 versions, so we should drop 25, right @econchick ?

**Review by DanSimon — COMMENTED** [review]
One potential thing, otherwise looks good

File: exec/tox.ini (hunk 1 above)
Comment (econchick) [reply exec/tox.ini:2]: Yes, that's true. Drop `25` here and the entire line 15 (for this tox and the other)

**Review by econchick — CHANGES_REQUESTED** [review]

**Review by fallonchen — APPROVED** [review]
:shipit:

**Pull request merged**